### PR TITLE
chore: simplify validation

### PR DESCRIPTION
**Pull request by onx2**

### CODE REVIEW

File: packages/solid/src/validation/url.ts [hunk 1]
```diff
@@ -12,9 +12,9 @@ type Value = MaybeValue<string>;
 export function url(error: string): (value: Value) => string {
   return (value: Value) => {
     try {
-      value && new URL(value);
+      new URL(value);
```
Comment (onx2) [inline packages/solid/src/validation/url.ts:15]: This should be sufficient I think?

```ts
// All of these will throw an error and be caught in the catch
new URL(true);
new URL({});
new URL([]);
new URL(42);
new URL("0");
new URL("false");
new URL(new Date());
new URL(-42);
new URL(12n);
new URL(3.14);
new URL(-3.14);
new URL(Infinity);
new URL(-Infinity); 
```

File: packages/solid/src/validation/url.ts [hunk 2]
```diff
@@ -12,9 +12,9 @@ type Value = MaybeValue<string>;
...
       return '';
-    } catch (_) {
+    } catch {
```
Comment (onx2) [inline packages/solid/src/validation/url.ts:17]: The `exceptionvar` [is optional ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#exceptionvar)

File: packages/solid/src/validation/url.ts (hunk 1 above)
Comment (fabian-hiller) [reply packages/solid/src/validation/url.ts:15]: Values are optional by default with Modular Forms, so `value && ...` is required.

File: packages/solid/src/validation/url.ts (hunk 2 above)
Comment (fabian-hiller) [reply packages/solid/src/validation/url.ts:17]: Good catch!

**Comment by fabian-hiller** [comment]
Since I have "paused" the work on Modular Forms and only change the code for very important things, I am not sure if I will merge this PR. I still plan to rewrite the library at some point, but that will take some time.

File: packages/solid/src/validation/url.ts (hunk 1 above)
Comment (onx2) [reply packages/solid/src/validation/url.ts:15]: `new URL(undefined)` should also throw as expected right?

**Comment by onx2** [comment]
> Since I have "paused" the work on Modular Forms and only change the code for very important things, I am not sure if I will merge this PR. I still plan to rewrite the library at some point, but that will take some time.

Ah that is unfortunate to hear because it looks like just what I'd want to use. Should I consider another library for production use?

File: packages/solid/src/validation/url.ts (hunk 1 above)
Comment (fabian-hiller) [reply packages/solid/src/validation/url.ts:15]: But we don't want to throw it in this case. Without `required()` there should be no error for missing or empty strings.

**Comment by fabian-hiller** [comment]
I recommend it for Qwik. For all other frameworks it depends on your use case. For example, with SolidJS there is no integration yet with SolidStart when using actions.

File: packages/solid/src/validation/url.ts (hunk 1 above)
Comment (onx2) [reply packages/solid/src/validation/url.ts:15]: Ah I see what you mean, thanks for the explanation

**Pull request closed without merging**